File: 240313/k번째로 신기한 문자열/kth-special-string.java
import java.io.*;
import java.util.*;

public class Main {
    public static int n, k;
    public static String T;
    public static List<String> words = new ArrayList<String>();
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        k = sc.nextInt();
        T = sc.next();
        sc.nextLine();
        while (n-- > 0) {
            String word = sc.nextLine();
            if (word.startsWith(T)) {
                words.add(word);
            }
        }
        // Collections.sort(words);
        words.sort(null);
        System.out.println(words.get(k-1));
    }
}

File: 240313/k번째로 신기한 문자열/kth-special-string.py
n, k, T = tuple(input().split())
n, k = (int(n), int(k))
words = [input() for _ in range(n)]

filtered_words = []

for word in words:
    if word.startswith(T):
        filtered_words.append(word)

filtered_words.sort()
print(filtered_words[k-1])